index.d.ts

```diff
@@ -1,426 +1,339 @@
-  version: string | undefined,
-  enabled: boolean
-}
+  // https://www.elastic.co/guide/en/apm/agent/nodejs/current/configuration.html
+  export interface AgentConfigOptions {
+    abortedErrorThreshold?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+    active?: boolean;
+    addPatch?: KeyValueConfig;
+    apiKey?: string;
+    apiRequestSize?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+    apiRequestTime?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+    asyncHooks?: boolean;
+    breakdownMetrics?: boolean;
+    captureBody?: CaptureBody;
+    captureErrorLogStackTraces?: CaptureErrorLogStackTraces;
+    captureExceptions?: boolean;
+    captureHeaders?: boolean;
+    captureSpanStackTraces?: boolean;
+    cloudProvider?: string;
+    configFile?: string;
+    containerId?: string;
+    disableInstrumentations?: string | string[];
+    disableSend?: boolean;
+    environment?: string;
+    errorMessageMaxLength?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+    errorOnAbortedRequests?: boolean;
+    filterHttpHeaders?: boolean;
+    frameworkName?: string;
+    frameworkVersion?: string;
+    globalLabels?: KeyValueConfig;
+    hostname?: string;
+    ignoreMessageQueues?: Array<string>;
+    ignoreUrls?: Array<string | RegExp>;
+    ignoreUserAgents?: Array<string | RegExp>;
+    instrument?: boolean;
+    instrumentIncomingHTTPRequests?: boolean;
+    kubernetesNamespace?: string;
+    kubernetesNodeName?: string;
+    kubernetesPodName?: string;
+    kubernetesPodUID?: string;
+    logLevel?: LogLevel;
+    logUncaughtExceptions?: boolean;
+    logger?: PinoLogger | Logger;
+    maxQueueSize?: number;
+    metricsInterval?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+    metricsLimit?: number;
+    payloadLogFile?: string;
+    centralConfig?: boolean;
+    sanitizeFieldNames?: Array<string>;
+    secretToken?: string;
+    serverCaCertFile?: string;
+    serverTimeout?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+    serverUrl?: string;
+    serviceName?: string;
+    serviceNodeName?: string;
+    serviceVersion?: string;
+    sourceLinesErrorAppFrames?: number;
+    sourceLinesErrorLibraryFrames?: number;
+    sourceLinesSpanAppFrames?: number;
+    sourceLinesSpanLibraryFrames?: number;
+    spanFramesMinDuration?: string;
+    stackTraceLimit?: number;
+    transactionIgnoreUrls?: Array<string>;
+    transactionMaxSpans?: number;
+    transactionSampleRate?: number;
+    useElasticTraceparentHeader?: boolean;
+    usePathAsTransactionName?: boolean;
+    verifyServerCert?: boolean;
+  }
 
-interface Taggable {
```

Comment:
REVIEWER NOTE: interfaces `Taggable` and `StartSpanFn` served no purpose. They are not meaningful to external users. They are not constructs in the actual code, and the type definitions here weren't actually using the inheritance: the `setLabel` et al functions were still defined on the interfaces above.